src/c_kzg_4844.h

```diff
@@ -36,6 +36,19 @@ extern "C" {
 // Macros
 ///////////////////////////////////////////////////////////////////////////////
 
+#ifndef FIELD_ELEMENTS_PER_BLOB
+#error FIELD_ELEMENTS_PER_BLOB is undefined. This value must be externally supplied.
+#endif // FIELD_ELEMENTS_PER_BLOB
+/**
+ * There are only 1<<32 2-adic roots of unity in the field, limiting the
+ * possible values of FIELD_ELEMENTS_PER_BLOB. The restriction to 1<<31 is a
+ * current implementation limitation. Notably, the size of the FFT setup would
+ * overflow uint32_t, which would casues issues
+ */
+#if (FIELD_ELEMENTS_PER_BLOB <= 0) || ((FIELD_ELEMENTS_PER_BLOB) > (1 << 31))
```

Comment:
For consistency, I'd either add or remove parentheses. I hope the user always provides a single integer value for this.

```suggestion
#if ((FIELD_ELEMENTS_PER_BLOB) <= 0) || ((FIELD_ELEMENTS_PER_BLOB) > (1 << 31))
```

or 

```suggestion
#if (FIELD_ELEMENTS_PER_BLOB <= 0) || (FIELD_ELEMENTS_PER_BLOB > (1 << 31))
```